Troubleshooting: can't log in (#17558)
metabase/metabase: docs/troubleshooting-guide/cant-log-in.md

# People can't log in to Metabase

> You are supposed to be able to log in to Metabase, but:

Reviewer: ```suggestion
You should be able to log in to Metabase, but:
```

- you can't see the login page, or
- your credentials aren't accepted.

> ## Are you using the right URL?

Reviewer: ```suggestion
## Are you using the right URL for your Metabase?
```

- Do you have the port as well as the host?
- Has someone moved the Metabase instance (e.g., it was in dev and now it's in production)?

> ## Are the credentials stored by Metabase or is the account managed using Single Sign-On?

Reviewer: ```suggestion
## Does your Metabase use Single Sign-On (SS0) to manage accounts?
```

- Other steps depend on knowing this

> - FIXME: can people look at their own profile to find out how their credentials are managed?

Reviewer: The tell here should be obvious when they sign in. They should have an option to use the third party identity provider, or "sign in with email".

- FIXME: where in the Admin Panel can someone find this?

> Note Metabase can't re-set passwords for SSO: if (for example) someone is logging in using their Google ID, they must re-set the Google password associated with that account.

Reviewer: ```suggestion
Metabase can't reset passwords for SSO: if (for example) someone is logging in using their Google ID, they must reset the Google password associated with that account.
```

> ## If the credentials are stored in Metabase, has the account been deactivated?

Reviewer: ```suggestion
## If they sign in by email (not by SSO), has their Metabase account been deactivated?
```

- Metabase doesn't delete accounts, but admins can deactivate them
- Go to Admin Panel > People
- If there is no "Deactivated" tab, there are no deactivated accounts
- If there is a "Deactivated" tab, look for the user in question
- Click on the "recycle" loop arrow to reactivate the account

## If the credentials are stored in Metabase, are they using the correct ID and password?

- [This FAQ][reset-password] explains how to re-set your own password
- Admins can also go to Admin Panel > People > "..." > "Reset password"

> ## If the account is managed by SSO, is SSO enabled?

Reviewer: We can probably skip this: if it's managed by SSO, then it's enabled.

For obvious reasons, regular users can't reactivate deactivated accounts. If you're an administrator and you want to do this for someone else:

1. Go to **Admin Panel** and select **People**.
2. If no **Deactivated** tab is available, there are no deactivated accounts, so this isn't the problem.
3. If there *is* a **Deactivated** tab, look for the user who isn't able to log in.
4. Click on the recycle loop arrow to reactivate the account.

## If you're logging in using Single Sign-On...

Metabase can't reset passwords for SSO: for example, if you are logging in using your Google ID, Google is managing your password, not Metabase, and for obvious security reasons, Google won't let other applications reset people's passwords.

If you are an admin and want to check SSO settings, go to **Admin Panel**, choose **Settings**, then select the **Authentication** tab.  [This FAQ][auth] explains how to configure SSO for various providers.

## If you're logging in via LDAP, is LDAP configured correctly?

**Root cause**: The LDAP connection is not configured correctly.

**Steps to take:**

1.  Go to the **Admin Panel** and choose **Authentication**.
2.  Make sure that LDAP is enabled

> 3.  Make sure Metabase has the correct host, port, and login credentials for your LDAP server. You can test this by logging into LDAP directly using some other application, such as [Apache Directory Studio][ads].

Reviewer 2: this is the link, just for reference: https://directory.apache.org/studio/

# People can't log in to Metabase

You should be able to log in to Metabase, but:

- you can't see the login page, or
- your credentials aren't accepted.

## Are you using the right URL for your Metabase?

**Root cause:** the Metabase instance you are trying to log in to isn't where you think it is or isn't accessible.

**Steps to take:**

1. Check whether you need to include a port number as well as a hostname in the connection URL. For example, Metabase might be at `https://example.com:3000/` instead of `https://example.com/`.
   - If you're an administrator, you'll have configured this.
   - If you're not, please ask your admin.

> 2. Check whether your Metabase instance has moved. For example, if you were using a trial instance of Metabase, but you're now in production, the URL might have changed.

Reviewer 3: I don't think I have seen more than a couple having this problem over the years.

## Does your Metabase use Single Sign-On (SS0) to manage accounts?

> Metabase can manage accounts itself, or administrators can configure it to let people sign on using third-party credentials. You can tell what kind of account you have when you try to sign in: if your login ID is your email address, Metabase is managing the account, but if you're using the option to sign in which a third-party identity provider, someone else is responsible for authenticating you.

Reviewer 3: I would remove this. It's also not correct - LDAP would be an example.

Author: Hm - @jeff-bruemmer thought this was OK. As a regular user, how can I tell if my credentials are in Metabase or via SSO?

Reviewer 3: The user would not know that unless they login. And the admin cannot really see that either depending on how the user was initially created.

Author: We should chat...

> ## If your account is managed by Metabase...

Reviewer 2: I would say here: If your instance does not use any SSO or something like that... as maybe some people won't know what might be managed by Metabase?

**Steps to take:**

For obvious reasons, regular users can't reactivate deactivated accounts. If you're an administrator and you want to do this for someone else:

1. Go to **Admin Panel** and select **People**.
2. If no **Deactivated** tab is available, there are no deactivated accounts, so this isn't the problem.
3. If there *is* a **Deactivated** tab, look for the user who isn't able to log in.
4. Click on the recycle loop arrow to reactivate the account.

### Do you need to reset your password?

**Root cause:** You have forgotten your password.

**Steps to take:**

1. If you are using the desktop Mac App, click on the **Help** menu item and select `Reset Password`.
2. If you are an administrator and want to reset someone's passsword, go to **Admin Panel**, select **People**, click on the ellipsis "..." next to the person's account, and select `Reset Password`.
3. If you are using the web app as a normal user:
   1. Click the link in the lower-right of the login screen that reads, "I seem to have forgotten my password".
   2. If your Metabase administrator has [set up email][setting-up-email] you will receive a password reset email.

>    3. If email has not been configured, you will need to contact a Metabase admin to perform a password reset.

Reviewer 3: Would probably move this to the top - it's the most common problem.

Author: done - thanks.
